main.tf

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  tmp_credentials_path = "${path.module}/terraform-google-credentials.json"
+  cache_path           = "${path.module}/cache/${var.platform}"
+  gcloud_tar_path      = "${local.cache_path}/google-cloud-sdk.tar.gz"
+  gcloud_bin_path      = "${local.cache_path}/google-cloud-sdk/bin"
+  components           = join(" ", var.additional_components)
+
+  gcloud  = "${local.gcloud_bin_path}/gcloud"
+  gsutil  = "${local.gcloud_bin_path}/gsutil"
+  bq      = "${local.gcloud_bin_path}/bq"
+  kubectl = "${local.gcloud_bin_path}/kubectl"
+
+  wait = length(null_resource.additional_components.*.triggers) + length(
+    null_resource.gcloud_auth_service_account_key_file.*.triggers,
+  ) + length(null_resource.gcloud_auth_google_credentials.*.triggers)
+}
+
+resource "null_resource" "decompress" {
+  triggers = {
+    always = uuid()
+  }
+
+  provisioner "local-exec" {
+    when    = create
+    command = "tar -xzf ${local.gcloud_tar_path} -C ${local.cache_path}"
+  }
+
+  provisioner "local-exec" {
```

Comment:
These destroy provisioners don't appear to be necessary. Is there a motivation for including them?

Reply:
I'm not quite sure, I was able to successfully run my tests without them, but I wasn't on the team who created this module, so let me try to find out.

Reply:
Talked to the creator of this module and it's a holdover from an old version of Terraform when you'd get an error if you didn't include both on create and on destroy provisioners when using local-exec. Now it's just a safety/habit, but we can probably remove them if we're requiring TF 12 and up.